src/main/kotlin/org/lunakoly/quicklink/actions/CopyLineLinkAction.kt

```diff
@@ -50,7 +50,22 @@ class CopyLineLinkAction : DumbAwareAction() {
         val currentFile = event.dataContext.getData(CommonDataKeys.VIRTUAL_FILE)?.canonicalFile
             ?: throw NoActiveFileException()
 
-        val lineNumber = 1 + editor.document.getLineNumber(editor.caretModel.offset)
+        val isSelection = editor.selectionModel.hasSelection()
+
+        val lineNumber: Pair<Int, Int>
+        val columnNumber: Pair<Int, Int>
+        if (isSelection) {
+            val startLine = 1 + editor.selectionModel.selectionStartPosition!!.getLine()
+            val endLine = 1 + editor.selectionModel.selectionEndPosition!!.getLine()
+            val startColumn = 1 + editor.selectionModel.selectionStartPosition!!.getColumn()
+            val endColumn = 1 + editor.selectionModel.selectionEndPosition!!.getColumn()
+            lineNumber = Pair(startLine, endLine)
+            columnNumber = Pair(startColumn, endColumn)
+        } else {
+            lineNumber = Pair(1 + editor.document.getLineNumber(editor.caretModel.offset), 0)
+            columnNumber = Pair(0, 0)
+        }
```

Comment:
Updated to use the `Selection` class here instead of `Pair`

Reply:
Closing this thread, as since there's only one variable (`selection`) and it can now be assigned as `val selection = if (...) { ... } else { ... }`, moving the code to a separate function wouldn't make it easier to read